docusaurus/docs/Android/02-client/06-guides/01-push-notifications/01-overview.mdx

```diff
@@ -126,39 +175,145 @@ class MyNotificationHandler(private val context: Context) : NotificationHandler
     }
 }
 ```
+</TabItem>
+
+<TabItem value="java" label="Java">
+
+```java
+class MyNotificationHandler implements NotificationHandler {
+
+    NotificationManager notificationManager;
+
+    public MyNotificationHandler(Context context) {
+        notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE) ;
+    }
+
+    @Override
+    public void onNotificationPermissionStatus(@NonNull NotificationPermissionStatus status) {
+        switch (status) {
+            case REQUESTED:
+                // invoked when POST_NOTIFICATIONS permission is requested
+                break;
+            case GRANTED:
+                // invoked when POST_NOTIFICATIONS permission is granted
+                break;
+            case DENIED:
+                // invoked when POST_NOTIFICATIONS permission is denied
+                break;
+            case RATIONALE_NEEDED:
+                // invoked when POST_NOTIFICATIONS permission requires rationale
+                break;
+        }
+    }
+
+    @Override
+    public void showNotification(@NonNull Channel channel, @NonNull Message message) {
+        Notification notification = new NotificationCompat.Builder(context, notificationChannelId)
+                .build();
+        notificationManager.notify(notificationId, notification);
+    }
+
+    @Override
+    public void dismissChannelNotifications(@NonNull String channelType, @NonNull String channelId) {
+        // Dismiss all notification related with this channel
+    }
+
+    @Override
+    public void dismissAllNotifications() {
+        // Dismiss all notifications
+    }
+
+    @Override
+    public boolean onChatEvent(@NonNull NewMessageEvent event) {
```

Comment:
Don't know, maybe worth adding to kotlin snippets

Reply:
It has a default implementation in Kotlin but it might be a good idea. We'll take a look on that during PN docs refactor